nvflare/app_opt/xgboost/histogram_based_v2/sec/client_handler.py

```diff
@@ -32,13 +32,28 @@
     generate_keys,
     split,
 )
+from nvflare.app_opt.xgboost.histogram_based_v2.sec.dam import DamDecoder
 from nvflare.app_opt.xgboost.histogram_based_v2.sec.data_converter import FeatureAggregationResult
-from nvflare.app_opt.xgboost.histogram_based_v2.sec.processor_data_converter import ProcessorDataConverter
+from nvflare.app_opt.xgboost.histogram_based_v2.sec.processor_data_converter import (
+    DATA_SET_HISTOGRAMS,
+    ProcessorDataConverter,
+)
 from nvflare.app_opt.xgboost.histogram_based_v2.sec.sec_handler import SecurityHandler
 
+try:
+    import tenseal as ts
+    from tenseal.tensors.ckksvector import CKKSVector
+
+    from nvflare.app_opt.he import decomposers
+    from nvflare.app_opt.he.homomorphic_encrypt import load_tenseal_context_from_workspace
+
+    tenseal_imported = True
+except Exception:
+    tenseal_imported = False
+
 
 class ClientSecurityHandler(SecurityHandler):
-    def __init__(self, key_length=1024, num_workers=10):
+    def __init__(self, key_length=1024, num_workers=10, tenseal_context_file="client_context.tenseal"):
```

Comment:
What if we just extend this class to have an HE version?
And just overwrite `_process_before_all_gather_v` and `_process_after_all_gather_v` methods?

Reply:
The logic is intertwined between two scheme (Vertical and horizontal). Separate classes may be confusing.